Zend/zend.c

```diff
@@ -175,8 +180,13 @@ static ZEND_INI_MH(OnSetExceptionStringParamMaxLen) /* {{{ */
 
 static ZEND_INI_MH(OnUpdateFiberStackSize) /* {{{ */
 {
+	zend_string *errstr;
 	if (new_value) {
-		EG(fiber_stack_size) = zend_atol(ZSTR_VAL(new_value), ZSTR_LEN(new_value));
+		EG(fiber_stack_size) = zend_ini_parse_quantity(new_value, &errstr);
+		if (errstr) {
+			zend_error(E_WARNING, "Invalid \"%s\" setting: %s", ZSTR_VAL(entry->name), ZSTR_VAL(errstr));
```

Comment:
Should we have a variant that also emits the error to avoid repeating this code? It also wouldn't need the temporary error string, which would simplify the API for the common use case where the warning is emitted.